pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -1510,87 +1511,30 @@ private BrokerResponseNative processLiteralOnlyQuery(long requestId, PinotQuery
   }
 
   // TODO(xiangfu): Move Literal function computation here from Calcite Parser.
-  private void computeResultsForExpression(Expression e, List<String> columnNames,
-      List<DataSchema.ColumnDataType> columnTypes, List<Object> row) {
-    if (e.getType() == ExpressionType.LITERAL) {
-      computeResultsForLiteral(e.getLiteral(), columnNames, columnTypes, row);
-    }
-    if (e.getType() == ExpressionType.FUNCTION) {
-      if (e.getFunctionCall().getOperator().equals("as")) {
-        String columnName = e.getFunctionCall().getOperands().get(1).getIdentifier().getName();
-        computeResultsForExpression(e.getFunctionCall().getOperands().get(0), columnNames, columnTypes, row);
-        columnNames.set(columnNames.size() - 1, columnName);
+  private void computeResultsForExpression(Expression expression, String[] columnNames, ColumnDataType[] columnTypes,
+      Object[] values, int index) {
+    if (expression.getType() == ExpressionType.LITERAL) {
+      computeResultsForLiteral(expression.getLiteral(), columnNames, columnTypes, values, index);
+    }
+    if (expression.getType() == ExpressionType.FUNCTION) {
```

Comment:
nit: AFAIU `computeResultsForLiteral` cannot change `expressiong.getType()`. Am I right? In that case can we add a `else` to this `if (expression.getType() == ExpressionType.FUNCTION) {` to improve readability?

Reply:
Good point.